presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -367,97 +354,91 @@ public static boolean isHiveNull(byte[] bytes)
+                return NullableValue.of(decimalType, longDecimalPartitionKey(value, decimalType, partitionName));
             }
         }
 
-        try {
-            if (HIVE_BOOLEAN.equals(hiveType)) {
-                if (isNull) {
-                    return NullableValue.asNull(BOOLEAN);
-                }
-                if (value.isEmpty()) {
-                    return NullableValue.of(BOOLEAN, false);
-                }
-                return NullableValue.of(BOOLEAN, parseBoolean(value));
+        if (BOOLEAN.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(BOOLEAN);
             }
-
-            if (HIVE_BYTE.equals(hiveType) || HIVE_SHORT.equals(hiveType) || HIVE_INT.equals(hiveType)) {
-                if (isNull) {
-                    return NullableValue.asNull(INTEGER);
-                }
-                if (value.isEmpty()) {
-                    return NullableValue.of(INTEGER, 0L);
-                }
-                return NullableValue.of(INTEGER, parseLong(value));
+            if (value.isEmpty()) {
+                return NullableValue.of(BOOLEAN, false);
             }
+            return NullableValue.of(BOOLEAN, booleanPartitionKey(value, partitionName));
+        }
 
-            if (HIVE_LONG.equals(hiveType)) {
-                if (isNull) {
-                    return NullableValue.asNull(BIGINT);
-                }
-                if (value.isEmpty()) {
-                    return NullableValue.of(BIGINT, 0L);
-                }
-                return NullableValue.of(BIGINT, parseLong(value));
+        if (INTEGER.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(INTEGER);
+            }
+            if (value.isEmpty()) {
+                return NullableValue.of(INTEGER, 0L);
             }
+            return NullableValue.of(INTEGER, integerPartitionKey(value, partitionName));
+        }
 
-            if (HIVE_DATE.equals(hiveType)) {
-                if (isNull) {
-                    return NullableValue.asNull(DATE);
-                }
-                return NullableValue.of(DATE, parseHiveDate(value));
+        if (BIGINT.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(BIGINT);
+            }
+            if (value.isEmpty()) {
+                return NullableValue.of(BIGINT, 0L);
             }
+            return NullableValue.of(BIGINT, bigintPartitionKey(value, partitionName));
+        }
 
-            if (HIVE_TIMESTAMP.equals(hiveType)) {
-                if (isNull) {
-                    return NullableValue.asNull(TIMESTAMP);
-                }
-                return NullableValue.of(TIMESTAMP, parseHiveTimestamp(value, timeZone));
+        if (DATE.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(DATE);
             }
+            return NullableValue.of(DATE, datePartitionKey(value, partitionName));
```

Comment:
Because we hadn't pass the presto Type parameter. So we had to duplicate all the parsing twice. Now  instead of passing `HiveType` i just pass the Presto `Type`. So i can reuse the partition key parsing methods.